project/cmake/scripts/common/projectmacros.cmake

```diff
@@ -50,10 +50,29 @@ endfunction()
 # Copied from FindGTest.cmake 
 # Thanks to Daniel Blezek <blezek@gmail.com> for the GTEST_ADD_TESTS code
 function(GTEST_ADD_TESTS executable extra_args)
+    get_property(XBMC_GENERATED_TEST_FILES GLOBAL PROPERTY XBMC_GENERATED_TEST_FILES)
+
     if(NOT ARGN)
         message(FATAL_ERROR "Missing ARGN: Read the documentation for GTEST_ADD_TESTS")
     endif()
     foreach(source ${ARGN})
+        get_property(GENERATED SOURCE "${source}" PROPERTY GENERATED)
+        get_property(SKIP_SCAN_TESTS SOURCE "${source}" PROPERTY SKIP_SCAN_TESTS)
+        if (NOT EXISTS "${source}")
+          # We can't rely on the GENERATED property here since it is likely that
+          # the file was added in a separate subdirectory. According to the CMake
+          # documentation, properties on source files can only be read by other
+          # listfiles in the same directory.
+          list(FIND XBMC_GENERATED_TEST_FILES "${source}" INDEX)
+          if ("${INDEX}" EQUAL -1)
```

Comment:
I've added a note about `set_source_files_properties` (which seems to be what is used internally)